pkg/clusteragent/admission/mutate/autoinstrumentation/injection_filter.go

```diff
@@ -0,0 +1,133 @@
+package autoinstrumentation
+
+import (
+	"fmt"
+	"sync"
+
+	mutatecommon "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate/common"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	apiServerCommon "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/common"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	// We need a global variable to store the filter instance
+	// because other webhooks depend on it.
+	//
+	// The "config" and the "tags" webhooks depend on the "auto_instrumentation"
+	// configuration to decide if a pod should be injected.
+	//
+	// They first check if the pod has the label to enable mutations.
+	// If it doesn't, they mutate the pod if the option to "mutate_unlabeled" is
+	// set to true or if APM SSI is enabled in the namespace.
+	autoInstrumentationInjectionFilterInit sync.Once
+	autoInstrumentationFilter              *containers.Filter
+	autoInstrumentationFilterError         error
+)
+
+// ShouldInject returns true if Admission Controller should inject standard tags, APM configs and APM libraries
```

Comment:
This function is refactored to re-use `ShouldMutatePod` as well as pass in its own additional checks, mainly `IsEnabledInNamespace` and the config checks for unlablled mutation. (prme-7)